test/data/certs/generate-test-certs

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash -xe
+
+# server certificate common name (hostname)
+SERVER_CN=${1:-server.example.com}
+
+# client certificate common name (hostname, uuid)
+CLIENT_CN=${2:-client.example.com}
+
+SUBJECT="/C=US/ST=CA/O=Example.com"
+CA_CN="Example CA"
+DAYS=9999
+PASSCA=pass:temporary_password
+PASSSV=pass:temporary_password
+PASSCT=pass:temporary_password
+
+# test-ca.crt
+openssl genrsa -passout $PASSCA -des3 -out test-ca.key 4096
+openssl req -passin $PASSCA -new -x509 -days $DAYS \
+  -key test-ca.key -out test-ca.crt -subj "$SUBJECT/CN=${CA_CN}"
+openssl x509 -purpose -in test-ca.crt
+openssl x509 -in test-ca.crt -out test-ca.pem -outform PEM
+
+# server.crt
+openssl genrsa -passout $PASSSV -des3 -out $SERVER_CN-server.key 4096
+openssl req -passin $PASSSV -new -key $SERVER_CN-server.key -out server.csr \
+  -addext "subjectAltName = DNS:${SERVER_CN}" \
+  -subj "$SUBJECT/CN=${SERVER_CN}"
+openssl x509 -req -passin $PASSCA -extfile /etc/pki/tls/openssl.cnf \
+  -extensions usr_cert -days $DAYS -in server.csr \
+  -extensions SAN -extfile <(cat /etc/pki/tls/openssl.cnf <(printf "\n[SAN]\nsubjectAltName=DNS:${SERVER_CN}\n")) \
```

Comment:
(nitpick) This will fail on e.g. Debian/Ubuntu which uses a different path for the ssl config,

Reply:
Yeah I only tested the script on Fedora, it is not just this there will be plenty of other differences I think